src/adapters/index.js

```diff
@@ -0,0 +1,30 @@
+import * as chirpstack from './chirpstack.js';
+import errors from '../errors.js';
+
+const adapters = {
+    chirpstack
+};
+
+// fastify force all headers in lowercase
+const headerName = 'ns-adapter';
+
+
+export default function adaptData ( request, reply ) {
```

Comment:
is it the correct form for the Fastify hook?
as I can see from docs there are variants
`fastify.addHook('preValidation', (request, reply, done) => {`
`fastify.addHook('preValidation', async (request, reply) => {`